eth_utils/string.py

```diff
@@ -1,15 +1,131 @@
-import functools
 import codecs
+import functools
+import itertools
+import json
+import warnings
+
+import eth_utils
```

Comment:
pretty sure this isn't great. but couldn't find a better solution - since utils imported from `web3.py` depend on functions in `hexidecimal.py` (`decode_hex`/`encode_hex` . . .). would it be better to combine `string.py` and `hexidecimal.py`?